### PR TITLE
NPM TypeScript type definitions

### LINKED ISSUE #27: Produce a typings file for TypeScript users
We should produce a typings file for our externally facing APIs

### PR DESCRIPTION
**Pull request by mhoeger**
This change adds a README and package.json.

The plan is to publish everything committed under `types/public`

Resolves #27

### CODE REVIEW

File: test/InterfacesTest.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Test typescript interfaces for ts compliation errors
-import * as azf from "../types/public/Interfaces";
+import { IFunction, IContext, IRequest} from "../types/public/Interfaces";
```
Comment (mhoeger) [inline test/InterfacesTest.ts]: changes to this file aren't really relevant, just trying a different (more common) way of importing and rearranged order of functions

File: types/public/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (mhoeger) [inline types/public/package.json]: A nested package.json, my plan here is to publish everything committed under `types/public`

File: types/public/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Type definitions for Azure Functions
+These are type definitions for using TypeScript with Azure Functions.
+
+Although Azure Functions does not directly support functions written in TypeScript, running transpiled functions is supported. Read more [here](https://docs.microsoft.com/azure/azure-functions/functions-reference-node#configure-function-entry-point) on configuring the location and name of your Azure Function.
```
Comment (ColbyTresness) [inline types/public/README.md]: This is probably fine for now, but you're framing this in a fairly negative light. Once we have better tooling support I'd change to something like "We support typescript development, but we don't automatically transpile and run TS files"

File: types/public/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+export { index };
+```
+
+# Versions
+Versioning of @azure/functions is tied to the version of the [Azure Functions Node.js Worker](https://github.com/Azure/azure-functions-nodejs-worker/releases) the types were generated from. You can find the Azure Functions Node.js Worker version of a given Function Runtime Version [here](https://github.com/Azure/azure-functions-host/releases). It is recommended that you take the latest version available.
+
+# Getting Started with Azure Functions
+For developing Azure Functions to run on Node.js, it is recommended that you use Visual Studio Code as your IDE. You can follow this tutorial to [create and deploy your first JavaScript function](https://docs.microsoft.com/azure/azure-functions/functions-create-first-function-vs-code).
```
Comment (ColbyTresness) [inline types/public/README.md]: Wording of the first clause is a bit awkward

File: types/public/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+The [Azure Functions developer guide](https://docs.microsoft.com/azure/azure-functions/functions-reference) and the[JavaScrip-specific developer guide](https://docs.microsoft.com/azure/azure-functions/functions-reference-node) are good resources for understanding more involved development.
```
Comment (ColbyTresness) [inline types/public/README.md]: To be pedantic, they're not gaining understanding more involved development, but they're gaining understanding of more involved Azure Functions concepts/more involved development with Functions. just rework slightly

**Comment by mhoeger** [comment]
@ColbyTresness - good feedback, thank you!! mind taking another look?

**Comment by ColbyTresness** [comment]
Looks great to me!

**Comment by jmshal** [comment]
Is there any ETA for publishing this package to npm? It would be great to have an official types package for Node.js Azure Functions.

**Pull request merged**